walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,50 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Comparator;
+import java.util.Arrays;
+import java.util.Iterator;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+  public Collection<TimeRange> query (Collection<Event> events, MeetingRequest request) {
+      Collection<String> people = request.getAttendees();
+      Collection<TimeRange> availableTimes = new ArrayList<TimeRange>();
+
+      ArrayList<TimeRange> bookedTimes = new ArrayList<TimeRange>();
+      getbookedTimes(events, people, bookedTimes);
+      Collections.sort(bookedTimes, TimeRange.ORDER_BY_START);
+      createAvailableRange(bookedTimes, availableTimes, request.getDuration());
+
+      
+      return availableTimes;
+  }
+
+  private void getbookedTimes (Collection<Event> events, Collection<String> people, ArrayList<TimeRange> bookedTimes) {
+      for(Event event: events){
+          if(!Collections.disjoint(event.getAttendees(), people)){
+            bookedTimes.add(event.getWhen());
+          }
+      }
+  }
+
+  private void createAvailableRange (ArrayList<TimeRange> bookedTimes, Collection<TimeRange> availableTimes, long duration) {
```

Comment:
let's return availableTimes by value as well. Dealing with values is simpler than output parameters, and it makes it more clear that this is a function that outputs available times given a set of inputs.

Reply:
done